src/deluge/gui/ui/browser/browser.cpp

```diff
@@ -1472,14 +1478,33 @@ ActionResult Browser::buttonAction(hid::Button b, bool on, bool inCardRoutine) {
 		if (on && !currentUIMode) {
 			return backButtonAction();
 		}
-	}
-	else {
+	} else if (b == Y_ENC) {
+		if (on && currentUIMode == UI_MODE_NONE) {
+			int32_t numItems = fileItems.getNumElements();
+
+			int32_t jumpCount = validFileItemsCount > 30 ? 30 : validFileItemsCount;
+
+			int32_t nextFileIndexSelected = random() % (jumpCount - 1);
+			int32_t offset = nextFileIndexSelected - fileIndexSelected;
+			fileIndexSelected = nextFileIndexSelected;
+			while (offset != 0) {
+				if (offset < 0) {
+					offset += 1;
+					selectEncoderAction(-1);
```

Comment:
I know this is gross. Every other method I tried would result in a corrupted file list. The only choice I had was to pretend that the user scrolled for some random number of times.